src/sync.rs

```diff
@@ -281,6 +286,137 @@ impl<T: StableDeref> FrozenVec<T> {
     // TODO add more
 }
 
+/// Append-only threadsafe version of `std::vec::Vec` where
+/// insertion does not require mutable access.
+/// Does not have locks, only allows `Copy` types and will
+/// spinlock on contention. The spinlocks are really rare as
+/// they only happen on reallocation due to a push going over
+/// the capacity.
+pub struct LockFreeFrozenVec<T: Copy> {
+    data: AtomicPtr<T>,
+    len: AtomicUsize,
+    cap: AtomicUsize,
+}
+
+impl<T: Copy> Drop for LockFreeFrozenVec<T> {
+    fn drop(&mut self) {
+        let ptr = self.aquire();
```

Comment:
issue: please comment why this is safe (because we have an exclusive reference)

Reply:
This made me realize we don't need to use atomic ops, with exclusive references we can just access the internals directly